### PR TITLE
Add creation of npmrc

### PR DESCRIPTION
**Pull request by ddgenome**
Fetch npm registries from graph, find configured registries, and
extract token and scope to create .npmrc.

Does this look right @cdupuis ?

### CODE REVIEW

File: codegen.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 schema: graphql/schema.json
-documents: node_modules/@atomist/skill/graphql/**/*.graphql
+documents: graphql/**/*.graphql
```
Comment (cdupuis) [inline codegen.yml:2]: Why did you move the predefined subscriptions into this repository? Was that to be able to generate types alongside your own queries?

I think we should provide types for those subscriptions from `@atomist/skill` to avoid that copying. WDYT? I could add them to the `subscription` package.

File: lib/npm.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		.filter(rp => configProviders[rp].typeName === "NpmJSRegistryProvider")
+		.map(rp => configProviders[rp].selectedResourceProviders)
+		.map(registries => registries.map(registry => registry.id))
+		.reduce((acc, cur) => acc.concat(cur), []) // flatten
+		.filter((value, index, self) => self.indexOf(value) === index); // unique
+
+	const npmJss = await ctx.graphql.query<
+		NpmJsRegistryProviderQuery,
+		NpmJsRegistryProviderQueryVariables
+	>({
```
Comment (cdupuis) [inline lib/npm.ts:42]: I think that can be simplified to `ctx.graphql.query<>("NpmJSRegistryProvider.graphql", {})`.

**Review by cdupuis — COMMENTED** [review]
A question and one comment.

File: codegen.yml (hunk 1 above)
Comment (ddgenome) [reply codegen.yml:2]: Yes. I tried setting documents to `{.,node_modules/@atomist/skill}/graphql/**/*.graphql`, but that did not work. It seemed that whatever was doing the globbing did not do the right thing.

It would be nice to have a standard schema.json and types in the skill project.

File: lib/npm.ts (hunk 2 above)
Comment (ddgenome) [reply lib/npm.ts]: I needed to key the types in there to avoid ending up with `any`, but I changed the arguments.

File: codegen.yml (hunk 1 above)
Comment (cdupuis) [reply codegen.yml:2]: Yes, this is in the works.

**Review by cdupuis — APPROVED** [review]
